### PR TITLE
feat(bigquery): implement ops.StructColumn, ops.ArrayColumn, ops.NthValue

### PR DESCRIPTION
**Pull request by krzysztof-kwitt**

### CODE REVIEW

File: ibis/backends/tests/test_window.py [hunk 1]
```diff
@@ -103,8 +103,13 @@ def calc_zscore(s):
             ),
             id="nth",
             marks=[
-                pytest.mark.notimpl(["bigquery", "pandas", "snowflake"]),
+                pytest.mark.notimpl(["pandas", "snowflake"]),
                 pytest.mark.notyet(["impala", "mssql"]),
+                pytest.mark.broken(
+                    "bigquery",
+                    strict=False,
+                    reason="output types is float64 instead of the expected float32",
```
Comment (cpcloud) [inline ibis/backends/tests/test_window.py:111]: Interesting, it sounds like there's maybe some missing type conversion code in BigQuery's `.execute` method.

Comment (cpcloud) [reply ibis/backends/tests/test_window.py:111]: In any case, that's a good follow up.

**Review by cpcloud — APPROVED** [review]
Thank you! I ran the tests locally and confirmed that the ones that you enable are now passing!

**Comment by krzysztof-kwitt** [comment]
Now, BigQuery backend is #3 in terms of the number of supported operations.
Screenshot 2023-01-06 at 18 33 05

**Pull request merged**